src/main/java/payapal/Perfection.java

```diff
@@ -0,0 +1,42 @@
+package main.java.payapal;
+
+public class Perfection {
+
+    private static Perfection perf;
+
+    public synchronized static Perfection getPerf() {
+        if (perf == null) {
+            perf = new Perfection();
+        }
+        return perf;
+    }
+
+
+    public static boolean isPerfect(long candidate) {
+        boolean retVal;
+        long[] divisors = GetDivisors(candidate);
+        int sum = 0;
+        for (int d = 0; d < 1000; d++)
```

Comment:
This for is going to iterate 1000 times no matter the size of divisors array. It means it will drive the app to an ArrayOutOfIndexException when it tries to access an inexisting position.  
Here are two options:  
- Limit the for with array size: `for (int d = 0; d < divisors.length; d++)`
- Change the for structure to `for(long divisor: divisors)`